### ISSUE TITLE
Time zones cause date reporting discrepancy when filtering vs grouping

### ISSUE THREAD

**Issue by bnygard** [metabase/metabase#3744]
When getting a rowcount based on the creation date of an object, filtering by specific creation date shows a different result than grouping by day of creation. The numbers are the same, but offset by a day. Switching MB reporting time zone to different options seems to have no effect.

![screen shot 2016-11-08 at 12 04 51 pm](https://cloud.githubusercontent.com/assets/6145345/20116274/7f0013b6-a5b0-11e6-9ef4-3f0a6ae2287a.png)
![screen shot 2016-11-08 at 12 04 15 pm](https://cloud.githubusercontent.com/assets/6145345/20116275/7f00fc36-a5b0-11e6-9ca1-209c791bad70.png)

Browser version: Chrome OS X 54.0.2840.71 (64-bit)
Browser TZ: PST

MB version: 0.20.3
MB report TZ: UTC

DB type: Postgres
DB TZ: UTC
Server TZ: UTC
Sample DB timestamp: "created_at: Tue, 08 Nov 2016 20:25:25 UTC +00:00"

**Comment by sorin** [comment]
I believe this might be a very similar issue, here's some steps to reproduce it (Postgres 9.6)

`CREATE TABLE test (created_at timestamp with time zone);`

`INSERT INTO test VALUES ('2015-02-01'), ('2015-02-02');`

`SET TIME ZONE 'America/Los_Angeles';`

`SELECT created_at::date, COUNT(*) FROM test GROUP BY created_at::date;`

```
 created_at | count 
------------+-------
 2015-02-02 |     1
 2015-02-01 |     1

```

<img width="597" alt="screenshot 2017-04-06 10 38 57" src="https://cloud.githubusercontent.com/assets/60167/24768635/240fe33c-1ab8-11e7-8ad0-ef334d73b0dd.png">

<img width="531" alt="screenshot 2017-04-06 11 00 14" src="https://cloud.githubusercontent.com/assets/60167/24768660/3d18d0c8-1ab8-11e7-8cb5-ca321ece91dd.png">

Even more interestingly, running this query:

`SELECT created_at::date, created_at::date::text||'-', COUNT(*) FROM test  GROUP BY created_at::date;`

produces this: 

<img width="431" alt="screenshot 2017-04-06 11 02 45" src="https://cloud.githubusercontent.com/assets/60167/24768756/9e06235e-1ab8-11e7-812a-d34924aa918b.png">

this maybe indicates that MB is trying to convert a string that looks like a date and apply the time zone transformation after getting the results form the database, but the source date is already in the target timezone.

**Comment by alee792** [comment]
Having the same issue as above. When grouping by a timestamp, MB converts an already converted timezone!

**Comment by Varunram** [comment]
Similar: #4180

**Comment by salsakran** [comment]
@sorin I'm having a hard time re-proing.

What timezones are your server and your browser in for the above tests?

eg, what I see on my dev machine -
![screen shot 2017-07-31 at 7 13 20 pm](https://user-images.githubusercontent.com/107439/28806365-66f4774a-7624-11e7-93cd-185b1515bd86.png)

**Comment by senior** [comment]
@bnygard @sorin @alee792 This was fixed in https://github.com/metabase/metabase/pull/5729 and released with 0.26 RC1. More information can be found on the [0.26.RC1 release page](https://github.com/metabase/metabase/releases/tag/v0.26.0.RC1). Would be great to get some validation on the fix if you're still experiencing the problem.
